wrappers/jira_client.py

```diff
@@ -25,23 +26,29 @@ def convert_seconds_to_jira_time(seconds):
 
         return " ".join(jira_time) if jira_time else "0s"
 
-    def __init__(self, config):
+    def __init__(self, config: dict):
         self.jira_url = config.get('jira_url')
-        self.tempo_api = config.get('tempo_api')
-        self.tempo_user = config.get('tempo_user')
-        self.tempo_password = config.get('tempo_password')
-        self.session = self.init_session()
+        self.tempo_url = config.get('tempo_url')
+        self.worklog_url = urljoin(self.tempo_url, 'worklogs')
+        self.jira_api_token = config.get('jira_api_token')
+        self.account_id = None
+        self.session = None
+        self.init_session()
 
     def init_session(self):
         session = requests.Session()
-        session.auth = (self.tempo_user, self.tempo_password)
-
-        api = self.jira_url + 'rest/api/2'
-        resp = session.get('%s/myself' % api)
-        if resp.ok:
-            return session
+        session.headers.update({
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {self.jira_api_token}"
+        })
+
+        resp = session.get(urljoin(self.jira_url, 'rest/api/3/myself'))
+        if resp.status_code == 200:
+            self.session = session
```

Comment:
This is for self._jira_session. 

For self.tempo_session, you cannot use the GET myself to check the token, but the way to use it is the same.